CONTRIBUTING.md

```diff
@@ -0,0 +1,24 @@
+# Contributing guidelines
+
+Thank you for wanting to contribute to this project!
+
+## Contributing code
+
+There are a couple of things we are going to look out for in PRs and knowing them upfront is going to reduce the number of times we will be going back and forth, making things more efficient.
+
+1. We have CI checks in place that validate formatting and code style.
+   Make sure `dprint check` and `cargo clippy` both finish without any warnings or errors on every commit.
+   If you don't already have `dprint` installed, you can obtain it in [various ways](https://dprint.dev/install/).
+2. All text documents (`CHANGELOG.md`, `README.md`, etc) should follow the [semantic linebreaks](https://sembr.org/) specification.
+3. We strive for atomic commits with good commit messages.
+   As an inspiration, read [this](https://chris.beams.io/posts/git-commit/) blogpost.
+   An atomic commit is a cohesive diff with formatting checks, linter and build passing.
+   Ideally, all tests are passing as well but we acknowledge that this is not always possible depending on the change you are making.
+4. If you are making any user-facing changes, include a changelog entry.
```

Comment:
Haha, does this mean only external contributors have to do this?

Reply:
we want to start doing that as well ;D